packages/core/upload/server/graphql.js

```diff
@@ -98,12 +97,29 @@ module.exports = ({ strapi }) => {
           },
 
           async resolve(parent, args) {
-            const { file: upload, info, ...fields } = args;
-
-            const file = await formatFile(upload, info, fields);
-            const uploadedFile = await getUploadService('upload').uploadFileAndPersist(file);
+            // create temporary folder to store files for stream manipulation
+            const tmpFolderPath = await fse.mkdtemp(path.join(os.tmpdir(), 'strapi-upload-'));
+            let sanitizedEntity;
+
+            try {
+              const { file: upload, info, ...fields } = args;
+
+              const file = await formatFile(upload, info, fields, { tmpFolderPath });
+              const uploadedFile = await getUploadService('upload').uploadFileAndPersist(
+                file,
+                {},
+                { tmpFolderPath }
+              );
+              sanitizedEntity = await toEntityResponse(uploadedFile, {
+                args,
+                resourceUID: fileTypeName,
+              });
+            } finally {
+              // delete temporary folder
+              await fse.remove(tmpFolderPath);
```

Comment:
Thank for being careful on that! Actually it will be ok because the function `fs.mkdtemp` (https://nodejs.org/api/fs.html#fspromisesmkdtempprefix-options) creates a unique folder by appending six random characters at the end. So the folder will be used only by the current request